plugins/router/f5/plugin.go

```diff
@@ -0,0 +1,183 @@
+package f5
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+	"github.com/golang/glog"
+
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+type F5Plugin struct {
+	F5Client *f5LTM
+}
+
+type F5PluginConfig struct {
+	Host     string
+	Username string
+	Password string
+	Vserver  string
+	Insecure bool
+}
+
+func NewF5Plugin(cfg F5PluginConfig) (*F5Plugin, error) {
+	f5LTMCfg := f5LTMCfg{
+		host:     cfg.Host,
+		username: cfg.Username,
+		password: cfg.Password,
+		vserver:  cfg.Vserver,
+		insecure: cfg.Insecure,
+	}
+	f5, err := newF5LTM(f5LTMCfg)
+	return &F5Plugin{f5}, err
+}
+
+func (p *F5Plugin) HandleEndpoints(eventType watch.EventType,
+	endpoints *kapi.Endpoints) error {
+
+	glog.V(4).Infof("Processing %d Endpoints for Name: %v (%v)",
+		len(endpoints.Subsets), endpoints.Name, eventType)
+
+	for i, s := range endpoints.Subsets {
+		glog.V(4).Infof("  Subset %d : %#v", i, s)
+	}
+
+	switch eventType {
+	case watch.Added, watch.Modified:
+		// Name of the pool in F5.
+		poolName := fmt.Sprintf("openshift_%s_%s_80",
+			endpoints.Namespace, endpoints.Name)
+
+		glog.V(4).Infof("Modifying endpoints for %s", poolName)
+
+		// The invocation of PoolHasMember has the side effect that it will
+		// initialise PoolMembers[poolName], which is used below, if necessary.
+		poolExists, err := p.F5Client.PoolExists(poolName)
```

Comment:
We can't rely on the route being created before the endpoints event reaches the plugin.